src/services/route-broadcaster.ts

```diff
@@ -406,11 +407,13 @@ export default class RouteBroadcaster {
       routingTableId: this.forwardingRoutingTable.routingTableId,
       currentEpoch: this.forwardingRoutingTable.currentEpoch,
       localRoutingTable: formatRoutingTableAsJson(this.localRoutingTable),
-      forwardingRoutingTable: formatRoutingTableAsJson(this.forwardingRoutingTable),
-      routingLog: this.forwardingRoutingTable.log.map(entry => ({
-        ...entry,
-        route: entry.route && formatRouteAsJson(entry.route)
-      })),
+      forwardingRoutingTable: formatForwardingRoutingTableAsJson(this.forwardingRoutingTable),
+      routingLog: this.forwardingRoutingTable.log
+        .filter(Boolean)
+        .map(entry => ({
+          ...entry,
+          route: entry && entry.route && formatRouteAsJson(entry.route)
```

Comment:
I think `entry && ` is unnecessary because of the preceding `.filter(Boolean)`.

Reply:
No, unfortunately, TypeScript isn't smart enough, so the `entry &&` is needed to make the type checker happy. :/